lib/pure/strutils.nim

```diff
@@ -531,6 +733,33 @@ proc split*(s: string, sep: string, maxsplit: int = -1): seq[string] {.noSideEff
   ## `split iterator <#split.i,string,string>`_.
   accumulateResult(split(s, sep, maxsplit))
 
+template accumulateResultReverse(iter: expr) =
+  # Similar to accumulateResult, but inserts into the result
+  # in reverse order
+  result = @[]
+  for x in iter: insert(result, x)
```

Comment:
What would you recommend doing in this case? Do you think it should just return in the order output by rsplit?